packages/ui/src/components/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FocusEventHandler, MouseEventHandler, ReactElement } from 'react';
+import { forwardRef, IconButton as ChakraIconButton, IconButtonProps as ChakraIconButtonProps } from '@chakra-ui/react';
+
+type IconButtonSize = 'sm' | 'md' | 'lg' | 'xs';
+
+type IconButtonColor = 'default' | 'brand' | 'danger';
+
+type IconButtonVariant = 'link' | 'outline' | 'solid' | 'ghost';
+
+export interface IconButtonProps {
+  icon: ReactElement;
+  /**
+   * Color variant of the icon button
+   */
+  color?: IconButtonColor;
+  /**
+   * Determines button's width, height and icon's size.
+   */
+  size?: IconButtonSize;
+  /**
+   * Style variant of the icon button
+   */
+  variant?: IconButtonVariant;
+  onClick?: MouseEventHandler;
+  onFocus?: FocusEventHandler;
+  onBlur?: FocusEventHandler;
+  'aria-label'?: string;
```

Comment:
For accessibility reasons, I would make this property obligatory. Additionally, it should have some description.